modules/smarthubBidAdapter.js

```diff
@@ -64,23 +68,56 @@ const getPlacementReqData = buildPlacementProcessingFunction({
 
 const buildRequests = (validBidRequests = [], bidderRequest = {}) => {
   const bidsByPartner = validBidRequests.reduce((bidsByPartner, bid) => {
-    const partner = getPartnerName(bid);
+    const partner = _getPartnerName(bid);
+    Object.assign(adapterState, ALIASES[partner]);
     (bidsByPartner[partner] = bidsByPartner[partner] || []).push(bid);
     return bidsByPartner;
   }, {});
   return Object.entries(bidsByPartner).map(([partner, validBidRequests]) => {
     return buildRequestsBase({
-      adUrl: _getUrl(partner),
+      adUrl: _getPartnerUrl(partner),
       bidderRequest,
       validBidRequests,
       placementProcessingFunction: getPlacementReqData
     })
   })
 }
 
+const getUserSyncs = (syncOptions, serverResponses, gdprConsent, uspConsent, gppConsent) => {
```

Comment:
instead of pulling this all out of utils, why not just post-process the utils output

Reply:
I haven't found any suitable utility to generate the getUserSyncs function with the required base url.
What do you mean by "post-process the utils output"?

Reply:
what about something like this import { getUserSyncs as baseSync } from '../libraries/teqblazeUtils/bidderUtils.js';

const getUserSyncs = (syncOptions, serverResponses, gdprConsent, uspConsent, gppConsent) => {
  const syncs = baseSync('')(syncOptions, serverResponses, gdprConsent, uspConsent, gppConsent);
  const syncUrl = SYNC_URLS[adapterState.area];
  const pid = adapterState.pid;

  return syncs.map(sync => ({
    ...sync,
    url: `${syncUrl}${sync.url}&pid=${pid}`
  }));
};


contxtful actually does something like this on your library https://github.com/prebid/Prebid.js/blob/f03b5cdb7e9973d82b3bd052e0c74f4894a953cd/modules/contxtfulBidAdapter.js#L9

Reply:
really great solution! thank you!